tensorboard/webapp/metrics/views/right_pane/settings_view_component.ts

```diff
@@ -71,7 +76,10 @@ export class SettingsViewComponent {
   @Input() stepMinMax!: {min: number; max: number};
 
   @Output() linkedTimeToggled = new EventEmitter<void>();
-  @Output() linkedTimeSelectionChanged = new EventEmitter<TimeSelection>();
+  @Output()
+  linkedTimeSelectionChanged = new EventEmitter<
+    TimeSelection & {source: RangeInputSource}
```

Comment:
This combined type and the RangeValuesEvent type are very similar. Could we use the RangeValuesEvent here? Maybe even change the RangeValuesEvent type to extend TimeSelection?

Reply:
I would rather the `range_input` component not know about linked time so I would prefer them use different objects.